csrc/scheduler/normalization_inner_outer.cpp

```diff
@@ -199,31 +439,111 @@ std::pair<int64_t, int64_t> getPersistentBufferSize(
   auto persistent_buffer_size_info = scheduler_utils::persistentBufferSize(
       fusion, runtime_info, persistent_buffer_info, data_cache);
 
-  // Note that projected buffer size can be zero
-  auto persistent_buffer_size =
-      persistent_buffer_size_info.projected_persistent_buffer_size == 0
-      ? persistent_buffer_size_info.persistent_buffer_size
-      : std::min(
-            persistent_buffer_size_info.persistent_buffer_size,
-            persistent_buffer_size_info.projected_persistent_buffer_size);
+  // Project to inputs when there is at least one outer broadcast tensor or
+  // projected persistent buffer size is smaller. When projecting to inputs, the
+  // outer broadcast tensor is reused in the loop over the iteration dimension,
+  // test shows it is faster than the non-projected version which requires
+  // reload from gmem for each iteration.
+  // Note: in current use cases (layer norm bwd and RMS norm bwd), there are
+  // outer broadcast tvs and always project to inputs.
+  const auto& outer_broadcast_tvs = getOuterBroadcastTvs(fusion, reduction_tvs);
+  bool can_project = ir_utils::getViewOps(fusion).empty() &&
+      persistent_buffer_size_info.projected_persistent_buffer_size > 0;
+  buffer_params.project_to_input = can_project &&
+      (!outer_broadcast_tvs.empty() ||
+       persistent_buffer_size_info.projected_persistent_buffer_size <
```

Comment:
it is calculated in `scheduler_utils::persistentBufferSize` based on the life scope of the persistent buffers.
It can be smaller than `persistent_buffer_size` and `0` indicates no buffer can be projected back to inputs.